rules/type-error.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+
+const tcIdentifiers = new Set([
```

Comment:
Some of these functions are methods on objects (like _.isFunction) , and some of them are independent functions (like isNaN).
I think there would be less false positives if we separated this list into two based on this difference. Some of theses items would appear in both lists (like for `isNaN` and `Number.isNaN` IIRC)